### ISSUE TITLE
Consider setting SO_REUSEADDR/SO_REUSEPORT

### ISSUE THREAD

**Issue by Erroneous1** [eddic/fastcgipp#50]
Using setsockopt SO_REUSEADDR (or SO_REUSEPORT if Linux >= 3.9, I'd assume yes these days) before bind/listen can make it so that a server that segfaults can start listening right away. I didn't need to worry about this on 2.1 because I used the spawn-fcgi program which I never got to segfault. A great description of these options and how they differ between operating systems it is on [stackoverflow](https://stackoverflow.com/a/14388707).

**Comment by eddic** [comment]
This sounds like a good idea to me based on what you've explained. Is there any reason we _wouldn't_ want to do this?

**Comment by Erroneous1** [comment]
There may be some security issues depending on your system. If you listen on a port without SO_REUSEADDR then nobody else can listen on that port (or at least shouldn't be able to). I assume if you have 2 processes able to bind/listen on the same port then sometimes accept() would return to one process and sometimes to the other.

I submitted a PR that lets you optionally set it with a default of current behavior. On my system SO_REUSEPORT isn't needed (Linux 4.16.13) though I didn't try it. I looked at boost::asio and they don't use appear to use SO_REUSEPORT for this.
